Clarification throughout the  abstract and introduction

rasbt opened the pull request:
Hi, all,

this is a really nice document listing a lot of interesting literature concerning deep learning and biomedicine. While I was reading it, made some notes about certain wordings that I found a bit awkward and added suggested re-wordings to this PR in hope it's going to be helpful.

agitter commented:
@rasbt Thanks for the suggestions and edits.

I'm going to leave this open for the time being.  We submitted the current version for re-review at a journal and are discussing what we want to do next with the manuscript in #810.  I'll return to this pull request once we make some progress in #810.

rasbt commented:
That makes sense and sounds like a good plan! Hope the re-review goes smoothly!

agitter commented:
@rasbt thanks again for your suggestions.  The manuscript has been provisionally accepted (#820) so we are in somewhat uncharted territory regarding adding new contributors.

I propose that we merge these changes and add you to the acknowledgements for your contributions.  Before proceeding, I wanted to confirm with you that sounds fair.  If so, @cgreene or I will review this pull request to comment on a couple specific lines.

rasbt commented:
Congratulations, that's great news! And sure, that's totally fine, I didn't expect to be e.g., a co-author for these relatively minor suggestions :)

cgreene commented on content/01.abstract.md:
```diff
@@ -1,11 +1,10 @@
+We examine applications of deep learning to a variety of biomedical problems---patient classification, fundamental biological processes, and treatment of patients---and discuss whether deep learning will be able to transform these tasks or if the biomedical sphere poses unique challenges.
+As a result from an extensive literature review, we find that deep learning has yet to revolutionize biomedicine or definitively resolve any of the most pressing challenges in the field, but promising advances have been made on the prior state of the art.
+Even though improvements over previous baselines have been modest, the recent progress indicates that deep learning methods will provide valuable means for speeding up or aiding human investigation.
+However, deep learning models are still regarded as black box algorithms, and more work is needed to address the common concerns related to interpretability and how to best model each problem.
```
I'm a bit reluctant to use the terminology `black box` here. I feel it's overused and under-justified, when many other methods that don't get this terminology applied have similar issues. What about:

> Though progress has been made in determining the primary factors that lead a specific deep neural network to make a specific prediction in a certain case, understanding how users should interpret these models to make specific mechanistic hypotheses remains an open challenge.

cgreene replied:
☝️ I'm not a huge fan of that, but I feel like the sentiment is what i'm going for so I'll put it there.

agitter replied:
I like the sentiment as well, but `determining the primary factors that lead a specific deep neural network to make a specific prediction in a certain case` is quite verbose

cgreene replied:
What about this:

> Though progress has been made linking a specific neural networks prediction to input features, understanding how users should interpret these models to make testable hypotheses about the system under study remains an open challenge.

agitter replied:
👍 but add `'` in `network's`

cgreene commented on content/01.abstract.md:
```diff
@@ -1,11 +1,10 @@
-We find that deep learning has yet to revolutionize or definitively resolve any of these problems, but promising advances have been made on the prior state of the art.
-Even when improvement over a previous baseline has been modest, we have seen signs that deep learning methods may speed or aid human investigation.
-More work is needed to address concerns related to interpretability and how to best model each problem.
+Deep learning, which describes a class of machine learning algorithms focussing on the training of deep artificial neural networks, has recently shown impressive results across a variety of domains.
```
I think we specifically left `artificial neural network` algorithms out of our definitions in many cases. What I think is more important is that these methods can work on raw data to produce intermediate features that are then used for some subsequent task. What about:

> Deep learning describes a class of machine learning algorithms that are capable of learning to combine relatively raw inputs into layers of intermediate features, and such algorithms now perform impressively across a variety of domains.

agitter replied:
@cgreene that version has good ideas, but I think it would work better as 2 sentences.  How about:

> Deep learning describes a class of machine learning algorithms that are capable of combining raw inputs into layers of intermediate features. These algorithms now perform impressively across a variety of domains.

I'm not sure why, but the longer `recently shown impressive results across a variety of domains` sounds more natural to me.

rasbt replied:
I agree with `recently shown impressive results across a variety of domains`. Another suggestion would be `produces remarkable results across a variety of domains compared to more traditional methods`, but I would prefer the former.

agitter replied:
Let's go with the more compact `recently shown impressive results across a variety of domains`

cgreene commented on content/01.abstract.md:
```diff
@@ -1,11 +1,10 @@
-Even when improvement over a previous baseline has been modest, we have seen signs that deep learning methods may speed or aid human investigation.
-More work is needed to address concerns related to interpretability and how to best model each problem.
+Deep learning, which describes a class of machine learning algorithms focussing on the training of deep artificial neural networks, has recently shown impressive results across a variety of domains.
+Biology and medicine are data-rich disciplines, but the data are complex and often ill-understood. Hence, deep learning techniques may be particularly well-suited to solve problems of these fields.
```
I don't really like this (or our initial framing). What about:

> Deep learning techniques may be particularly well suited for challenges in biology and medicine, which are data-rich disciplines where there is often a complex biological system between what we can measure and what we wish to know.

cgreene replied:
☝️ needs revision too, but I vaguely like the direction it's going in

agitter replied:
I'm okay with the @rasbt version here but am happy to switch it if you want.

cgreene commented on content/02.intro.md:
```diff
@@ -7,11 +7,11 @@ Automated algorithms that extract meaningful patterns could lead to actionable k
 The term _deep learning_ has come to refer to a collection of new techniques that, together, have demonstrated breakthrough gains over existing best-in-class machine learning algorithms across several fields.
 For example, over the past five years these methods have revolutionized image classification and speech recognition due to their flexibility and high accuracy [@doi:10.1038/nature14539].
-More recently, deep learning algorithms have shown promise in fields as diverse as high-energy physics [@doi:10.1038/ncomms5308], dermatology [@doi:10.1038/nature21056], and translation among written languages [@arxiv:1609.08144].
+More recently, deep learning algorithms have shown promise in fields as diverse as high-energy physics [@doi:10.1038/ncomms5308], computational chemistry [@doi:10.1002/jcc.24764], dermatology [@doi:10.1038/nature21056], and translation among written languages [@arxiv:1609.08144].
```
👍

agitter replied:
👍 

Also noting that we cite this review later in the intro.

cgreene commented on content/02.intro.md:
```diff
@@ -7,11 +7,11 @@ Automated algorithms that extract meaningful patterns could lead to actionable k
 
-Deep learning approaches grew from research in neural networks, which were first proposed in 1943 [@doi:10.1007/BF02478259] as a model for how our brains process information.
-The history of neural networks is interesting in its own right [@doi:10.1103/RevModPhys.34.135].
+Deep learning approaches grew from research on artificial neurons, which were first proposed in 1943 [@doi:10.1007/BF02478259] as a model for how the neurons in a biological brain process information.
```
👍

cgreene commented on content/02.intro.md:
```diff
@@ -26,11 +26,11 @@ In particular, deep learning approaches can be used both in *supervised* applica
 In fact, if the dataset is "labeled" with binary classes, a simple neural network with no hidden layers and no cycles between units is equivalent to logistic regression if the output layer is a sigmoid (logistic) function of the input layer.
-Similarly, for continuous outcomes, linear regression can be seen as a simple neural network.
-Thus, in some ways, supervised deep learning approaches can be seen as a generalization of regression models that allow for greater flexibility.
+Similarly, for continuous outcomes, linear regression can be seen as a single-layer neural network.
```
👍

cgreene commented on content/02.intro.md:
```diff
@@ -26,11 +26,11 @@ In particular, deep learning approaches can be used both in *supervised* applica
 Recently, hardware improvements and very large training datasets have allowed these deep learning techniques to surpass other machine learning algorithms for many problems.
 In a famous and early example, scientists from Google demonstrated that a neural network "discovered" that cats, faces, and pedestrians were important components of online videos [@url:http://research.google.com/archive/unsupervised_icml2012.html] without being told to look for them.
-What if, more generally, deep learning could solve the challenges presented by the growth of data in biomedicine? Could these algorithms identify the "cats" hidden in our data---the patterns unknown to the researcher---and suggest ways to act on them? In this review, we examine deep learning's application to biomedical science and discuss the unique challenges that biomedical data pose for deep l ...
+What if, more generally, deep learning take advantage of the growth of data in biomedicine to tackle challenges in this field? Could these algorithms identify the "cats" hidden in our data---the patterns unknown to the researcher---and suggest ways to act on them? In this review, we examine deep learning's application to biomedical science and discuss the unique challenges that biomedical data po ...
```
👍

agitter commented on content/01.abstract.md:
```diff
@@ -1,11 +1,10 @@
+Deep learning, which describes a class of machine learning algorithms focussing on the training of deep artificial neural networks, has recently shown impressive results across a variety of domains.
+Biology and medicine are data-rich disciplines, but the data are complex and often ill-understood. Hence, deep learning techniques may be particularly well-suited to solve problems of these fields.
+We examine applications of deep learning to a variety of biomedical problems---patient classification, fundamental biological processes, and treatment of patients---and discuss whether deep learning will be able to transform these tasks or if the biomedical sphere poses unique challenges.
+As a result from an extensive literature review, we find that deep learning has yet to revolutionize biomedicine or definitively resolve any of the most pressing challenges in the field, but promising advances have been made on the prior state of the art.
```
Could switch `As a result from` to `Following` to help the word count

agitter commented on content/01.abstract.md:
```diff
@@ -1,11 +1,10 @@
+Biology and medicine are data-rich disciplines, but the data are complex and often ill-understood. Hence, deep learning techniques may be particularly well-suited to solve problems of these fields.
+We examine applications of deep learning to a variety of biomedical problems---patient classification, fundamental biological processes, and treatment of patients---and discuss whether deep learning will be able to transform these tasks or if the biomedical sphere poses unique challenges.
+As a result from an extensive literature review, we find that deep learning has yet to revolutionize biomedicine or definitively resolve any of the most pressing challenges in the field, but promising advances have been made on the prior state of the art.
+Even though improvements over previous baselines have been modest, the recent progress indicates that deep learning methods will provide valuable means for speeding up or aiding human investigation.
```
Perhaps `have been modest in general`?  There have been a few impressive exceptions since we first drafted the abstract.

agitter commented on content/citation-tags.tsv:
```diff
@@ -80,6 +80,7 @@ Gerstein2016_scaling	doi:10.1186/s13059-016-0917-0
 Ghandi2014_enhanced	doi:10.1371/journal.pcbi.1003711
 Ghosh1992_sequence	doi:10.1117/12.140112
 Glorot2011_domain	url:http://citeseerx.ist.psu.edu/viewdoc/summary?doi=10.1.1.231.3442
+Goh2017_compchemistry	doi:10.1002/jcc.24764
```
We can remove the tag if the DOI is being using in the reference above. The tag is only needed when we use it directly in the citation from Markdown.

agitter commented on content/02.intro.md:
```diff
@@ -26,11 +26,11 @@ In particular, deep learning approaches can be used both in *supervised* applica
-Similarly, for continuous outcomes, linear regression can be seen as a simple neural network.
-Thus, in some ways, supervised deep learning approaches can be seen as a generalization of regression models that allow for greater flexibility.
+Similarly, for continuous outcomes, linear regression can be seen as a single-layer neural network.
+Thus, in some ways, supervised deep learning approaches can be seen as a extension of regression models that allow for greater flexibility.
```
`an extension`

rasbt replied:
maybe as an add-on to that sentence: `and are especially well-suited for modeling non-linear relationships among the input features.`

agitter replied:
👍

cgreene commented:
@rasbt : do you want to make the final changes and then we'll squash merge? I am happy to approve after @agitter's revisions. 😄

rasbt commented:
Sure, happy to do that and will get to it later this evening

rasbt commented:
Just addressed all the comments in the last commit. You probably need to trim the abstract quite a bit if it's a 200 and not 250 word limit -- but I think it's been already a bit longer prior to these changes ;)
